### ISSUE TITLE
useDevicesList - video facingMode not work

### ISSUE THREAD

**Issue by Ltyi** [vueuse/vueuse#1240]
```js
const { videoInputs: cameras } = useDevicesList({
  constraints: { video: { facingMode: { ideal: 'environment' } } }
});

// cameras.value[0]?.deviceId << this is front-facing camera
// cameras.value[1]?.deviceId << this is back camera
```

The back camera should be first elemet in cameras array

**Comment by webfansplz** [comment]
This may not be a bug, `useDevicesList` doesn't change the order of output .

Maybe you can use the Native API to see if you get different results .

**Comment by Ltyi** [comment]
> This may not be a bug, `useDevicesList` doesn't change the order of output .
> 
> Maybe you can use the Native API to see if you get different results .

yes, so I tried nativa api `navigator.mediaDevices.getUserMedia({ video: { facingMode: { ideal: 'environment' } }  })` to active correct camera.

but if I want to use `useUserMedia()` and `useDevicesList()` to active my back camera on iPhone or iPad, where should I put this config?

**Comment by webfansplz** [comment]
> > This may not be a bug, `useDevicesList` doesn't change the order of output .
> > Maybe you can use the Native API to see if you get different results .
> 
> yes, so I tried nativa api `navigator.mediaDevices.getUserMedia({ video: { facingMode: { ideal: 'environment' } } })` to active correct camera.
> 
> but if I want to use `useUserMedia()` and `useDevicesList()` to active my back camera on iPhone or iPad, where should I put this config?

I'm not sure about your question ?

1. `useDevicesList` does not work with this config 
2.  Just the order of output  diff from native API

**Comment by Ltyi** [comment]
> > > This may not be a bug, `useDevicesList` doesn't change the order of output .
> > > Maybe you can use the Native API to see if you get different results .
> > 
> > 
> > yes, so I tried nativa api `navigator.mediaDevices.getUserMedia({ video: { facingMode: { ideal: 'environment' } } })` to active correct camera.
> > but if I want to use `useUserMedia()` and `useDevicesList()` to active my back camera on iPhone or iPad, where should I put this config?
> 
> I'm not sure about your question ?
> 
> 1. `useDevicesList` does not work with this config
> 2. Just the order of output  diff from native API

Sorry, my question is `getUserMedia` can't use `facingMode` config, and `getDevicesList` either, so if I want use this config, the only way is `navigator.mediaDevices.getUserMedia` ?

**Comment by okxiaoliang4** [comment]
You can get the back camera device id and put it to`useUserMedia`

**Comment by Ltyi** [comment]
> You can get the back camera device id and put it to`useUserMedia`

thank you!

but `videoInputs` it seems no infomation to know which one is back camera?

**Comment by okxiaoliang4** [comment]
> > You can get the back camera device id and put it to`useUserMedia`
> 
> thank you!
> 
> but `videoInputs` it seems no infomation to know which one is back camera?

Maybe try assert `device.label`

**Comment by felwip** [comment]
Hi, I feel that I have got a similar issue. I am trying to set some `MediaStreamConstraints` on the video (`frameRate`, `height`, `width`) and on the video (`echoCancellation`, `autoGainControl`, `noiseSuppression`). Though I cannot find a way to pass these options to useUserMedia and apply them to the corresponding device.
Is it possible or could it be desired in the future to pass options to `useUserMedia` in a similar way as in `useDisplayMedia`?

Example use-case:
`useUserMedia(video: {deviceId: 'default', {constraints}}, audio: {deviceId: 'default', {constraints}} )`

**Comment by okxiaoliang4** [comment]
> Hi, I feel that I have got a similar issue. I am trying to set some `MediaStreamConstraints` on the video (`frameRate`, `height`, `width`) and on the video (`echoCancellation`, `autoGainControl`, `noiseSuppression`). Though I cannot find a way to pass these options to useUserMedia and apply them to the corresponding device.
> 
> Is it possible or could it be desired in the future to pass options to `useUserMedia` in a similar way as in `useDisplayMedia`?
> 
> 
> 
> Example use-case:
> 
> `useUserMedia(video: {deviceId: 'default', {constraints}}, audio: {deviceId: 'default', {constraints}} )`

PR welcome :)

**Comment by Gazoon007** [comment]
> > Hi, I feel that I have got a similar issue. I am trying to set some `MediaStreamConstraints` on the video (`frameRate`, `height`, `width`) and on the video (`echoCancellation`, `autoGainControl`, `noiseSuppression`). Though I cannot find a way to pass these options to useUserMedia and apply them to the corresponding device.
> > Is it possible or could it be desired in the future to pass options to `useUserMedia` in a similar way as in `useDisplayMedia`?
> > Example use-case:
> > `useUserMedia(video: {deviceId: 'default', {constraints}}, audio: {deviceId: 'default', {constraints}} )`
> 
> PR welcome :)

Has this issue been resolved in this PR cmiiw? https://github.com/vueuse/vueuse/pull/2032

**Comment by ilyaliao** [comment]
> > > Hi, I feel that I have got a similar issue. I am trying to set some `MediaStreamConstraints` on the video (`frameRate`, `height`, `width`) and on the video (`echoCancellation`, `autoGainControl`, `noiseSuppression`). Though I cannot find a way to pass these options to useUserMedia and apply them to the corresponding device.
> > > Is it possible or could it be desired in the future to pass options to `useUserMedia` in a similar way as in `useDisplayMedia`?
> > > Example use-case:
> > > `useUserMedia(video: {deviceId: 'default', {constraints}}, audio: {deviceId: 'default', {constraints}} )`
> > 
> > 
> > PR welcome :)
> 
> Has this issue been resolved in this PR cmiiw? [#2032](https://github.com/vueuse/vueuse/pull/2032)

Yes, I believe the requirements here have been met. I’ll close this issue. Thanks! :)
